src/github/mod.rs

```diff
@@ -131,59 +134,46 @@ pub fn ingest_since(repo: &str, start: DateTime<UTC>) -> DashResult<()> {
 pub fn update_issue(repo: &str, number: i32) -> DashResult<()> {
     let issue = GH.fetch_issue(repo, number)?;
     let comments = GH.fetch_comments(repo, number)?;
+    let conn = &*DB_POOL.get()?;
 
-    handle_issue(issue, repo)?;
+    handle_issue(conn, issue, repo)?;
     for comment in comments {
-        handle_comment(comment, repo)?;
+        handle_comment(conn, comment, repo)?;
     }
 
     Ok(())
 }
 
 pub fn update_pr(repo: &str, number: i32) -> DashResult<()> {
     let pr = GH.fetch_pr(repo, number)?;
-    handle_pr(pr, repo)
+    let conn = &*DB_POOL.get()?;
+
+    handle_pr(conn, pr, repo)
 }
 
-pub fn handle_pr(pr: PullRequestFromJson, repo: &str) -> DashResult<()> {
+pub fn handle_pr(conn: &PgConnection, pr: PullRequestFromJson, repo: &str) -> DashResult<()> {
     use domain::schema::pullrequest::dsl::*;
-
-    let conn = DB_POOL.get()?;
-
     if let Some(ref assignee) = pr.assignee {
-        handle_user(&*conn, assignee)?;
+        handle_user(conn, assignee)?;
     }
 
     let pr: PullRequest = pr.with_repo(repo);
-
-    let existing_id = pullrequest.select(id)
-        .filter(number.eq(&pr.number))
-        .filter(repository.eq(&pr.repository))
-        .first::<i32>(&*conn)
-        .ok();
-
-    if let Some(current_id) = existing_id {
-        diesel::update(pullrequest.find(current_id)).set(&pr).execute(&*conn)?;
-    } else {
-        diesel::insert(&pr).into(pullrequest).execute(&*conn)?;
-    }
-
+    diesel::insert(&pr.on_conflict((repository, number), do_update().set(&pr)))
+        .into(pullrequest).execute(conn)?;
     Ok(())
 }
 
-pub fn handle_comment(comment: CommentFromJson, repo: &str) -> DashResult<()> {
-    let conn = DB_POOL.get()?;
-
-    handle_user(&*conn, &comment.user)?;
+pub fn handle_comment(conn: &PgConnection, comment: CommentFromJson, repo: &str) -> DashResult<()> {
+    handle_user(conn, &comment.user)?;
 
     let comment: IssueComment = comment.with_repo(repo)?;
 
-    if issuecomment::table.find(comment.id).get_result::<IssueComment>(&*conn).is_ok() {
+    if issuecomment::table.find(comment.id).get_result::<IssueComment>(conn).is_ok() {
         diesel::update(issuecomment::table.find(comment.id)).set(&comment)
-            .execute(&*conn)?;
+            .execute(conn)?;
         Ok(())
     } else {
-        diesel::insert(&comment).into(issuecomment::table).execute(&*conn)?;
+        diesel::insert(&comment).into(issuecomment::table).execute(conn)?;
```

Comment:
`nag::update_nags` unconditionally adds RFC-Bot comments -- so to avoid double-posting, we only want to run it when we insert the comment, instead of updating [0]. I couldn't figure out how to get information from Diesel about whether the upsert was an insert or an update, so I just left the code as-is (I can't even figure out how to do it in raw Postgres without adding a dummy column or something 😝).


[0]  This probably has some issue where someone messes up the RFC-Bot syntax when posting, updates the comment to fix it, and then gets confused when RFCBot doesn't respond, but that's another story...